### PR TITLE
Add Hearthstone Cask

### LINKED ISSUE #30533: Cask request: Hearthstone
### Cask details

Name: Hearthstone

Homepage: http://us.battle.net/hearthstone/en/

Download URL: https://www.battle.net/download/getInstallerForGame?os=mac&locale=enUS&version=LIVE&gameProgram=HEARTHSTONE

Description:
Popular free card game said to rival magic the gathering

### PR DESCRIPTION
**Pull request by miccal**
After making all changes to the cask:

- [X] `brew cask audit --download {{cask_file}}` is error-free.
- [X] `brew cask style --fix {{cask_file}}` reports no offenses.
- [X] The commit message includes the cask’s name and version.

Additionally, **if adding a new cask**:

- [X] Named the cask according to the [token reference].
- [X] `brew cask install {{cask_file}}` worked successfully.
- [X] `brew cask uninstall {{cask_file}}` worked successfully.
- [X] Checked there are no [open pull requests] for the same cask.
- [X] Checked the cask was not already refused in [closed issues].
- [X] Checked the cask is submitted to [the correct repo].

Closes https://github.com/caskroom/homebrew-cask/issues/30533.

### CODE REVIEW

File: Casks/hearthstone.rb [hunk 1]
```diff
@@ -0,0 +1,10 @@
+cask 'hearthstone' do
+  version :latest
+  sha256 :no_check
+
+  url 'https://www.battle.net/download/getInstallerForGame?os=mac&locale=enUS&version=LIVE&gameProgram=HEARTHSTONE'
+  name 'Hearthstone'
+  homepage 'http://us.battle.net/hearthstone/en/'
+
+  app 'Hearthstone-Setup.app'
```
Comment (vitorgalvao) [inline Casks/hearthstone.rb]: Looks like this is a manual installer, from the name.

**Comment by miccal** [comment]
@vitorgalvao you are correct.

@kaungst since you requested this Cask, are you able to provide details of how to add an `uninstall` stanza?

**Comment by gobinathm** [comment]
@kaungst @miccal @vitorgalvao I believe we already have this [CASK](https://github.com/caskroom/homebrew-cask/blob/master/Casks/battle-net.rb)

To Confirm, indeed both the app are same I just executed a test.  
* Downloaded the app from about mentioned `Download URL: https://www.battle.net/download/getInstallerForGame?os=mac&locale=enUS&version=LIVE&gameProgram=HEARTHSTONE`
* Installed it manually
* After the installation got completed i tried removing the app using APPCLEANER & here is the screen shot
battle-net-cask
* Now if you take a look at uninstall section of Casks/battle-net.rb , you will notice both the same

(Just observation, i might be wrong)

**Comment by miccal** [comment]
@gobinathm thanks for that - I had noticed the Cask you linked to, and in the `url` is the sting `Program=BATTLENET_APP`, while in this PR the `url` has the string `Program=HEARTHSTONE`, so I assumed they were different?

**Comment by gobinathm** [comment]
@miccal initially even i thought that.  But after downloading & installing HEARTHSTONE, everything seems to be same in the App. I'm not seeing any difference

**Comment by miccal** [comment]
Thanks for the information @gobinathm.

Closing as a potential duplicate.

**Pull request closed without merging**